src/Plugin/AddHostPlugin.php

```diff
@@ -52,8 +52,11 @@ public function __construct(UriInterface $host, array $config = [])
     public function handleRequest(RequestInterface $request, callable $next, callable $first)
     {
         if ($this->replace || $request->getUri()->getHost() === '') {
-            $uri = $request->getUri()->withHost($this->host->getHost());
-            $uri = $uri->withScheme($this->host->getScheme());
+            $uri = $request->getUri()
+                ->withHost($this->host->getHost())
+                ->withScheme($this->host->getScheme())
+                ->withPort($this->host->getPort())
```

Comment:
according to the PSR interface, getPort should return null if its the default port for this protocol, and `withPort(null)` must remove the port so that the default port is used. with a very poor uri implementation, this could lead to changing http://localhost/ to http://localhost:80/ which is a bit ugly but not wrong. i prefer that over complicated logic in here to detect the default port ourselves.

we must set the port to null if it is null, for the case of `replace`, as the original request could have specified a different port.